checker: miss the check when we are working in raw pointer and interface

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Linux Mint 20.2
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i5-6300U CPU @ 2.40GHz
CC version: cc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0

getwd: /home/vincent/Github/vlang/v
vmodules: /home/vincent/.vmodules
vroot: /home/vincent/Github/vlang/v
vexe: /home/vincent/Github/vlang/v/v
vexe mtime: 2022-04-10 22:54:08
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 a0e7a46

Git version: git version 2.25.1
Git vroot status: weekly.2022.13-109-ga0e7a46b
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 3654d6a8

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg tmp.v`
```v
interface A {
    a int
    b() int
}

struct Ba {
    a int
}

fn (b &Ba) b() int {
    return b.a
}

fn main() {
    a := &Ba{a: 1}
    b := A(a)
}

```

**What did you expect to see?**

checker error that tell message about wrong cast type


**What did you see instead?**
```
tmp.v:16:5: warning: unused variable: `b`
   14 | fn main() {
   15 |     a := &Ba{a: 1}
   16 |     b := A(a)
      |     ^
   17 | }
==================
/tmp/v_1000/tmp.288171835460834715.tmp.c:11486: error: 'A' undeclared
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/parser/struct.v
+++ b/vlib/v/parser/struct.v
@@ -493,6 +493,11 @@ fn (mut p Parser) interface_decl() ast.InterfaceDecl {
 		return ast.InterfaceDecl{}
 	}
 	modless_name := p.check_name()
+	if modless_name.len == 1 && modless_name[0].is_capital() {
+		p.error_with_pos('single letter capital names are reserved for generic template types.',
+			name_pos)
+		return ast.InterfaceDecl{}
+	}
 	if modless_name == 'IError' && p.mod != 'builtin' {
 		p.error_with_pos('cannot register interface `IError`, it is builtin interface type',
 			name_pos)

--- a/vlib/v/parser/tests/interface_name_err.out
+++ b/vlib/v/parser/tests/interface_name_err.out
@@ -1,0 +1,5 @@
+vlib/v/parser/tests/interface_name_err.vv:1:11: error: single letter capital names are reserved for generic template types.
+    1 | interface A {
+      |           ^
+    2 |     a int
+    3 |     b() int

--- a/vlib/v/parser/tests/interface_name_err.vv
+++ b/vlib/v/parser/tests/interface_name_err.vv
@@ -1,0 +1,19 @@
+interface A {
+	a int
+	b() int
+}
+
+struct Ba {
+	a int
+}
+
+fn (b &Ba) b() int {
+	return b.a
+}
+
+fn main() {
+	a := &Ba{
+		a: 1
+	}
+	b := A(a)
+}

